### PR TITLE
Add persistence function for discovered schema

### PR DESCRIPTION
**Pull request by malikdiarra**
## What
This adds persistence functions for the schema tables added in https://github.com/airbytehq/airbyte/pull/10286

This PR is part of [#9895](https://github.com/airbytehq/airbyte/issues/9895)

### CODE REVIEW

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java [hunk 1]
```diff
@@ -1200,6 +1298,60 @@ private void writeStandardSyncState(final List<StandardSyncState> configs, final
...
+  }
+
+  private void writeActorCatalog(final List<ActorCatalog> configs, final DSLContext ctx) {
+    final OffsetDateTime timestamp = OffsetDateTime.now();
+    configs.forEach((actorCatalog) -> {
+      final boolean isExistingConfig = ctx.fetchExists(select()
+          .from(ACTOR_CATALOG)
+          .where(ACTOR_CATALOG.ID.eq(actorCatalog.getId())));
+
+      if (isExistingConfig) {} else {
```
Comment (tuliren) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java:1315]:
```suggestion
      if (!isExistingConfig) {
```

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java [hunk 2]
```diff
@@ -1200,6 +1298,60 @@ private void writeStandardSyncState(final List<StandardSyncState> configs, final
...
+  }
+
+  private void writeActorCatalogFetchEvent(final List<ActorCatalogFetchEvent> configs, final DSLContext ctx) {
+    final OffsetDateTime timestamp = OffsetDateTime.now();
+    configs.forEach((actorCatalogFetchEvent) -> {
+      final boolean isExistingConfig = ctx.fetchExists(select()
+          .from(ACTOR_CATALOG_FETCH_EVENT)
+          .where(ACTOR_CATALOG_FETCH_EVENT.ID.eq(actorCatalogFetchEvent.getId())));
+
+      if (isExistingConfig) {} else {
```
Comment (tuliren) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java:1341]:
```suggestion
      if (!isExistingConfig) {
```

File: airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: update migration description in the class name
```
Comment (tuliren) [inline airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java:19]:
```suggestion
```

File: airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  private static final Logger LOGGER = LoggerFactory.getLogger(
+      V0_35_28_001__AddActorCatalogMetadataColumns.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
```
Comment (tuliren) [inline airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java:31]:
```suggestion
```

**Review by tuliren — COMMENTED** [review]
Looks good in general. Could you add more unit tests for the `list-` methods in `DatabaseConfigPersistence`?

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java [hunk 5]
```diff
@@ -1200,6 +1298,60 @@ private void writeStandardSyncState(final List<StandardSyncState> configs, final
...
+  }
+
+  private void writeActorCatalogFetchEvent(final List<ActorCatalogFetchEvent> configs, final DSLContext ctx) {
+    final OffsetDateTime timestamp = OffsetDateTime.now();
+    configs.forEach((actorCatalogFetchEvent) -> {
+      final boolean isExistingConfig = ctx.fetchExists(select()
+          .from(ACTOR_CATALOG_FETCH_EVENT)
+          .where(ACTOR_CATALOG_FETCH_EVENT.ID.eq(actorCatalogFetchEvent.getId())));
+
+      if (!isExistingConfig) {
```
Comment (subodh1810) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java:1341]: Shouldnt we have an update feature like other configs?

File: airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class V0_35_28_001__AddActorCatalogMetadataColumns extends BaseJavaMigration {
```
Comment (subodh1810) [inline airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java:19]: Can we add a test for the migration?

**Comment by malikdiarra** [comment]
> Looks good in general. Could you add more unit tests for the `list-` methods in `DatabaseConfigPersistence`?

This is done!

**Review by subodh1810 — CHANGES_REQUESTED** [review]

File: airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java (hunk 6 above)
Comment (malikdiarra) [reply airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_28_001__AddActorCatalogMetadataColumns.java:20]: Done!

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java (hunk 5 above)
Comment (malikdiarra) [reply airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java:1341]: Those table are used to cache information, so insert and get should really be the only access patterns. I can add the update case for consistency though

Comment (malikdiarra) [reply airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java]: I added the update case, as well as deletion as well as dumpAllConfig and replaceAllConfig operations.

**Review by subodh1810 — APPROVED** [review]

**Pull request merged**